pkg/option/config.go

```diff
@@ -3381,6 +3385,24 @@ func (c *DaemonConfig) checkIPv6NativeRoutingCIDR() error {
 	return nil
 }
 
+func (c *DaemonConfig) checkIPAMDelegatedPlugin() error {
+	if c.IPAM == ipamOption.IPAMDelegatedPlugin {
+		// When using IPAM delegated plugin, IP addresses are allocated by the CNI binary,
+		// not the daemon. Therefore, features which require the daemon to allocate IPs for itself
+		// must be disabled.
+		if c.EnableIPv4 && c.LocalRouterIPv4 == "" {
+			return fmt.Errorf("LocalRouterIPv4 must be provided when IPv4 is enabled with IPAMDelegatedPlugin")
+		}
+		if c.EnableIPv6 && c.LocalRouterIPv6 == "" {
+			return fmt.Errorf("LocalRouterIPv6 must be provided when IPv6 is enabled with IPAMDelegatedPlugin")
+		}
+		if c.EnableEndpointHealthChecking {
+			return fmt.Errorf("EnableEndpointHealthChecking must be disabled with IPAMDelegatedPlugin")
+		}
```

Comment:
For more actionable and user-friendly error messages, I'd suggest to change these to explicitly mention the CLI option names, not the internal names:

```suggestion
		if c.EnableIPv4 && c.LocalRouterIPv4 == "" {
			return fmt.Errorf("--%s must be provided when IPv4 is enabled with --%s=%s", LocalRouterIPv4, IPAM, IPAMDelegatedPlugin)
		}
		if c.EnableIPv6 && c.LocalRouterIPv6 == "" {
			return fmt.Errorf("--%s must be provided when IPv6 is enabled with --%s=%s", LocalRouterIPv6, IPAM, IPAMDelegatedPlugin)
		}
		if c.EnableEndpointHealthChecking {
			return fmt.Errorf("--%s must be disabled with --%s=%s", EnableEndpointHealthChecking, IPAM, IPAMDelegatedPlugin)
		}
```